nixos/modules/services/matrix/draupnir.nix

```
@@ -0,0 +1,278 @@
+            '';
+          };
+          managementRoom = lib.mkOption {
+            type = lib.types.str;
+            default = "#moderators:example.org";
+            description = ''
+              The room ID or alias where moderators can use the bot's functionality.
+
+              The bot has no access controls, so anyone in this room can use the bot - secure this room!
+
+              Warning: When using a room alias, make sure the alias used is on the local homeserver!
+              This prevents an issue where the control room becomes undefined when the alias can't be resolved.
+            '';
+          };
+
+          protectedRooms = lib.mkOption {
+            type = lib.types.listOf lib.types.str;
+            default = [ ];
+            example = lib.literalExpression ''
+              [
+                "https://matrix.to/#/#yourroom:example.org"
+                "https://matrix.to/#/#anotherroom:example.org"
+              ]
+            '';
+            description = ''
+              A list of rooms to protect (matrix.to URLs).
+              These can also be configured interactively.
+
+              Note that this option does nothing in Draupnir v2+!
+            '';
+          };
+          #endregion
+        };
+      };
+    };
+    #endregion
+  };
+  #endregion
+
+  #region Service configuration
+  config = lib.mkIf cfg.enable {
+    assertions = [
+      {
+        assertion = !(cfg.pantalaimon.enable && cfg.pantalaimon.passwordFile == null);
+        message = "Specify pantalaimon.passwordFile";
+      }
+      {
+        assertion = !(cfg.pantalaimon.enable && cfg.accessTokenFile != null);
+        message = "Do not specify accessTokenFile when using pantalaimon";
+      }
+      {
+        assertion = !(!cfg.pantalaimon.enable && cfg.accessTokenFile == null);
+        message = "Specify accessTokenFile when not using pantalaimon";
+      }
+    ];
+
+    services.pantalaimon-headless.instances."draupnir" = lib.mkIf cfg.pantalaimon.enable (cfg.pantalaimon.options);
+
+    systemd.services.draupnir = {
+      description = "Draupnir - a moderation tool for Matrix";
+      requires = lib.optionals (cfg.pantalaimon.enable) [
+        "pantalaimon-draupnir.service"
+      ];
+      wants = [
+        "network-online.target"
+        "matrix-synapse.service"
+        "conduit.service"
+        "dendrite.service"
+      ];
+      after = [
+        "network-online.target"
+        "matrix-synapse.service"
+        "conduit.service"
+        "dendrite.service"
+      ];
+      wantedBy = [ "multi-user.target" ];
+
+      serviceConfig = {
+        ExecStart = utils.escapeSystemdExecArgs ([
+          (lib.getExe cfg.package)
+          "--draupnir-config" configFile
+        ] ++ lib.optionals (cfg.pantalaimon.enable && cfg.pantalaimon.passwordFile != null) [
+          "--pantalaimon-password-path"
+          "/run/credentials/draupnir.service/pantalaimon_password"
```

`man systemd.exec(5)` recommends using `${CREDENTIALS_DIRECTORY}` in `ExecStart=`.
```suggestion
          "\${CREDENTIALS_DIRECTORY}/pantalaimon_password"
```

---

Does that work with `escapeSystemdExecArgs`?